lib/GraphReader.cc

```diff
@@ -163,54 +165,80 @@ void GraphReader::read_partition(BipartiteGraph::ContainerType& vmap) {
 void GraphReader::read_preference_lists(BipartiteGraph::ContainerType& A, BipartiteGraph::ContainerType& B) {
     // read the lists
     while (curtok_ != TOK_AT) {
         // read the vertex for which the preference list is given
         std::string a = lexer_->get_lexeme();
-        match(TOK_STRING);
+        match(TOK_STRING); //consume the string because you have already stored it in a 
         match(TOK_COLON); // skip the colon
 
         // read and store the preference list
         auto& a_vertex = A[a];
         PreferenceList& pref_list = a_vertex->get_preference_list();
         PreferenceList& pref_list_lq = a_vertex->get_preference_list_lq();
+        int currank_ = 0;
         while (curtok_ != TOK_SEMICOLON) {
-            std::string b = lexer_->get_lexeme();
-            match(TOK_STRING);
-            const auto& b_vertex = B[b];
-            pref_list.emplace_back(b_vertex);
-
-            // Store critical vertices in two preference lists.
-            if (b_vertex->get_lower_quota() > 0)
-            {
-              pref_list_lq.emplace_back(b_vertex);
+            if (curtok_ == TOK_LEFT_BRACE) { //this means that ties are present 
+                match(TOK_LEFT_BRACE);
+                PreferenceList tied_list = PreferenceList();
+                while(curtok_ != TOK_RIGHT_BRACE){
+                    //store it as a preference list and set tie_map_[i] = preference list 
+                    std::string b = lexer_->get_lexeme();
+                    match(TOK_STRING);
+                    const auto& b_vertex = B[b];
+                    tied_list.emplace_back(b_vertex);
+                    // Store critical vertices in two preference lists.
+                    if (b_vertex->get_lower_quota() > 0)
+                    {
+                        pref_list_lq.emplace_back(b_vertex);
+                    }
+                    // if there are more vertices, they must
+                    // be delimited using commas
+                    if (curtok_ != TOK_RIGHT_BRACE) {
+                        match(TOK_COMMA);
+                    }
+                }
+                match(TOK_RIGHT_BRACE);
+                pref_list.set_ties(currank_, tied_list);
+                pref_list.emplace_back(nullptr);
+            }
+            else{
+                std::string b = lexer_->get_lexeme();
+                match(TOK_STRING);
+                const auto& b_vertex = B[b];
+                pref_list.emplace_back(b_vertex); //add b to the end of preference list 
+
+                // Store critical vertices in two preference lists.
+                if (b_vertex->get_lower_quota() > 0)
+                {
+                    pref_list_lq.emplace_back(b_vertex);
+                }
             }
-
             // if there are more vertices, they must
             // be delimited using commas
             if (curtok_ != TOK_SEMICOLON) {
                 match(TOK_COMMA);
             }
+            currank_++;
```

Comment:
Use ++currank. Although, there is no difference between ++currank and currank++, stylistically code uses the first form everywhere.